test/WebJobs.Script.Tests/BlobLeaseManagerTests.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public class BlobLeaseManagerTests
+    {
+        [Fact]
+        [Trait("Category", "E2E")]
+        public async Task HasLease_WhenLeaseIsAcquired_ReturnsTrue()
+        {
+            string connectionString = AmbientConnectionStringProvider.Instance.GetConnectionString(ConnectionStringNames.Storage);
+            string hostId = Guid.NewGuid().ToString();
+            var traceWriter = new TestTraceWriter(System.Diagnostics.TraceLevel.Verbose);
+
+            using (var manager = await BlobLeaseManager.CreateAsync(connectionString, TimeSpan.FromSeconds(15), hostId, traceWriter))
+            {
+                await TestHelpers.Await(() => manager.HasLease);
+
+                Assert.Equal(hostId, manager.LeaseId);
+            }
+        }
+
+        [Fact]
+        [Trait("Category", "E2E")]
+        public async Task HasLeaseChanged_WhenLeaseStateChanges_IsFired()
+        {
```

Comment:
It'd be good to have a test the other way as well -- make sure it fires when you lose a lease.